engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentCreateProcessor.java

```diff
@@ -85,26 +84,30 @@ public void processRecord(final TypedRecord<DeploymentRecord> command) {
 
     final DeploymentRecord deploymentEvent = command.getValue();
 
-    deploymentTransformer.transform(deploymentEvent);
+    final Either<Failure, Void> result = deploymentTransformer.transform(deploymentEvent);
 
-    try {
-      createTimerIfTimerStartEvent(command);
-    } catch (final RuntimeException e) {
-      final String reason = String.format(COULD_NOT_CREATE_TIMER_MESSAGE, e.getMessage());
-      responseWriter.writeRejectionOnCommand(command, RejectionType.PROCESSING_ERROR, reason);
-      rejectionWriter.appendRejection(command, RejectionType.PROCESSING_ERROR, reason);
-      return;
-    }
+    if (result.isRight()) {
```

Comment:
🔧 You could invert this if and fail early. This helps reduce the nesting of the rest of the code, which means the reader has less code/context to keep in mind.